packages/api/cms-api/src/warnings/warning.resolver.ts

```diff
@@ -25,10 +30,73 @@ export class WarningResolver {
     }
 
     @Query(() => PaginatedWarnings)
-    async warnings(@Args() { status, search, filter, sort, offset, limit }: WarningsArgs): Promise<PaginatedWarnings> {
-        const where = gqlArgsToMikroOrmQuery({ search, filter }, this.repository);
+    async warnings(
+        @Args() { status, search, filter, scopes, sort, offset, limit }: WarningsArgs,
+        @GetCurrentUser() user: CurrentUser,
+    ): Promise<PaginatedWarnings> {
+        // check if there are any scopes that the user does not have permission to
+        const allowedScopesForUser = user.permissions.find(({ permission }) => permission === "warnings")?.contentScopes;
+
+        for (const scope of scopes) {
+            if (!allowedScopesForUser?.find((allowedScope) => isEqual(allowedScope, scope))) {
+                throw new UnauthorizedException("Scopes were passed that the user does not have permission to");
```

Comment:
I believe HTTP exceptions don't work correctly in GraphQL resolvers. Maybe we should throw a normal error here?

Reply:
I didn't know that using HTTP exceptions changes `extensions.code`, which we could consider [here](https://github.com/vivid-planet/comet/blob/next/packages/admin/admin/src/error/errordialog/createErrorDialogApolloLink.tsx#L22-L30), so maybe using UnauthorizedException is a good idea after all. @fraxachun what do you think?

Reply:
@raphaelblum could you please change it back? Sorry 😬

Reply:
no problem, changed here: 983f7780291c1834bd0cd22141431199bd34ca68